exporting/mongodb/mongodb.c

```diff
@@ -0,0 +1,310 @@
+        free_bson(insert, connector_specific_data->last_buffer->documents_inserted);
+    }
+    insert = callocz((size_t)stats->chart_buffered_metrics, sizeof(bson_t *));
+    connector_specific_data->last_buffer->insert = insert;
+
+    BUFFER *buffer = (BUFFER *)instance->buffer;
+    char *start = (char *)buffer_tostring(buffer);
+    char *end = start;
+
+    size_t documents_inserted = 0;
+
+    while (*end && documents_inserted <= (size_t)stats->chart_buffered_metrics) {
+        while (*end && *end != '\n')
+            end++;
+
+        if (likely(*end)) {
+            *end = '\0';
+            end++;
+        } else {
+            break;
+        }
+
+        bson_error_t error;
+        insert[documents_inserted] = bson_new_from_json((const uint8_t *)start, -1, &error);
+
+        if (unlikely(!insert[documents_inserted])) {
+            error("EXPORTING: %s", error.message);
+            free_bson(insert, documents_inserted);
+            return 1;
+        }
+
+        start = end;
+
+        documents_inserted++;
+    }
+
+    buffer_flush(buffer);
+
+    connector_specific_data->last_buffer->documents_inserted = documents_inserted;
+    connector_specific_data->last_buffer = connector_specific_data->last_buffer->next;
+
+    return 0;
+}
+
+/**
+ * MongoDB connector worker
+ *
+ * Runs in a separate thread for every instance.
+ *
+ * @param instance_p an instance data structure.
+ */
+void mongodb_connector_worker(void *instance_p)
+{
+    struct instance *instance = (struct instance *)instance_p;
+    struct mongodb_specific_config *connector_specific_config = instance->config.connector_specific_config;
+    struct mongodb_specific_data *connector_specific_data =
+        (struct mongodb_specific_data *)instance->connector_specific_data;
+
+    while (!netdata_exit) {
+        struct stats *stats = &instance->stats;
+
+        uv_mutex_lock(&instance->mutex);
+        uv_cond_wait(&instance->cond_var, &instance->mutex);
+
+        bson_t **insert = connector_specific_data->first_buffer->insert;
+        size_t documents_inserted = connector_specific_data->first_buffer->documents_inserted;
+
+        connector_specific_data->first_buffer->insert = NULL;
+        connector_specific_data->first_buffer->documents_inserted = 0;
+        connector_specific_data->first_buffer = connector_specific_data->first_buffer->next;
+
+        uv_mutex_unlock(&instance->mutex);
+
+        size_t data_size = 0;
+        for (size_t i = 0; i < documents_inserted; i++) {
+            data_size += insert[i]->len;
+        }
+
+        debug(
+            D_BACKEND,
+            "EXPORTING: mongodb_insert(): destination = %s, database = %s, collection = %s, data size = %zu",
+            instance->config.destination,
+            connector_specific_config->database,
+            connector_specific_config->collection,
+            data_size);
+
+        if (unlikely(documents_inserted == 0))
+            continue;
+
+        bson_error_t error;
```

Comment:
I will do the same request here, I do not think to have a variable with the same name of a local function as a good thing, the motives were described before.

Reply:
Fixed